### ISSUE TITLE
Enhancement: generate realistic audio using engine-sim

### ISSUE THREAD

**Issue by RichardScharf** [stuntrally/stuntrally3#7]
Since there's no discussions page for enhancements I thought I'd create an Issue.

I'd like to suggest integrating the Engine Simulator from ange-yaghi to generate more realistic engine sounds for the game https://github.com/ange-yaghi/engine-sim.

It is written in C++ and currently under the MIT license.

**Comment by RichardScharf** [comment]
That's unfortunate, I wasn't aware that open source projects could have such OS dependency problems.

**Comment by cryham** [comment]
Okay that newer version (without sources)
https://github.com/Engine-Simulator/engine-sim-community-edition
worked on Windows. Starts on Wine but probably too much CPU use makes it jitter.

Well it would be best to have some way to set throttle to like 4 rpm amounts eg. 1000 2000 3000 4000 etc
or anyhow setting it with mouse not with the freaking Q,W,E,R keys, because it's otherwise impossible to target constant rpm.

**Comment by RichardScharf** [comment]
The community edition works flawlessly for me using Bottles as Flatpak with default settings:
 (Wine default for Gaming: Win64, Soda-7.0-9 container)

The CPU Usage maxes out for me at 80% for a single core (of 16 total)
my specs are:

**OS:** Linux Mint 21.2 x86_64
**DE:** Cinnamon 5.8.4
**Kernel:** 5.15.0-82-generic

**CPU:** Intel i9-9900K (16) @ 5.000GHz
**GPU:** NVIDIA GeForce GTX 1080
**Memory:** 3547MiB / 32018MiB

As for the RPM values
The instruction (readme) on the github page of the community edition says:

H | Enable RPM hold (see below for instructions)
-- | --
G + Scroll | Change hold speed
Space + Scroll | Fine throttle adjustment
1, 2, 3, 4, 5 | Simulation time warp

Pressing `H`, then `D` and then holding `G` while scrolling the `mouse wheel` lets me precisely fix rpm in place.

another way is to hold down `space` and scroll the `mouse wheel` which is more natural I guess.

**Comment by cryham** [comment]
Great. I didn't realize that was all possible.

**Comment by cryham** [comment]
Okay I did record sounds back then. Got stuck for a while since engine-sim doesn't simulate an exhaust muffler, and it's a notable difference in sound.
Now I did progress lately.
I was searching and found many graphs
https://duckduckgo.com/?q=muffler+transmission+loss+db&t=ftsa&atb=v395-1&iax=images&ia=images
like here
https://reference.wolfram.com/language/PDEModels/tutorial/Acoustics/ModelCollection/AcousticMuffler.html
which when done in Audacity by Curve EQ, get better results.
There are still limits in engine-sim, high frequency or the explosion sound is just basically noise and it's bad. Surely worse when compared to real sounds.
But I think I'll have new sounds made and definitely will replace that old VDrift sound.
